include/constants/battle.h

```diff
@@ -283,44 +283,39 @@
-#define B_WEATHER_RAIN_PRIMAL         (1 << 3)
-#define B_WEATHER_RAIN                (B_WEATHER_RAIN_TEMPORARY | B_WEATHER_RAIN_DOWNPOUR | B_WEATHER_RAIN_PERMANENT | B_WEATHER_RAIN_PRIMAL)
-#define B_WEATHER_SANDSTORM_TEMPORARY (1 << 4)
-#define B_WEATHER_SANDSTORM_PERMANENT (1 << 5)
-#define B_WEATHER_SANDSTORM           (B_WEATHER_SANDSTORM_TEMPORARY | B_WEATHER_SANDSTORM_PERMANENT)
-#define B_WEATHER_SUN_TEMPORARY       (1 << 6)
-#define B_WEATHER_SUN_PERMANENT       (1 << 7)
-#define B_WEATHER_SUN_PRIMAL          (1 << 8)
-#define B_WEATHER_SUN                 (B_WEATHER_SUN_TEMPORARY | B_WEATHER_SUN_PERMANENT | B_WEATHER_SUN_PRIMAL)
-#define B_WEATHER_HAIL_TEMPORARY      (1 << 9)
-#define B_WEATHER_HAIL_PERMANENT      (1 << 10)
-#define B_WEATHER_HAIL                (B_WEATHER_HAIL_TEMPORARY | B_WEATHER_HAIL_PERMANENT)
-#define B_WEATHER_STRONG_WINDS        (1 << 11)
-#define B_WEATHER_ANY                 (B_WEATHER_RAIN | B_WEATHER_SANDSTORM | B_WEATHER_SUN | B_WEATHER_HAIL | B_WEATHER_STRONG_WINDS | B_WEATHER_SNOW | B_WEATHER_FOG)
-#define B_WEATHER_PRIMAL_ANY          (B_WEATHER_RAIN_PRIMAL | B_WEATHER_SUN_PRIMAL | B_WEATHER_STRONG_WINDS)
-#define B_WEATHER_SNOW_TEMPORARY      (1 << 12)
-#define B_WEATHER_SNOW_PERMANENT      (1 << 13)
-#define B_WEATHER_SNOW                (B_WEATHER_SNOW_TEMPORARY | B_WEATHER_SNOW_PERMANENT)
-#define B_WEATHER_FOG_TEMPORARY       (1 << 14)
-#define B_WEATHER_FOG_PERMANENT       (1 << 15)
-#define B_WEATHER_FOG                 (B_WEATHER_FOG_TEMPORARY | B_WEATHER_FOG_PERMANENT)
-
-// Battle Weather as enum
-#define ENUM_WEATHER_NONE                 0
-#define ENUM_WEATHER_RAIN                 1
-#define ENUM_WEATHER_SUN                  2
-#define ENUM_WEATHER_SANDSTORM            3
-#define ENUM_WEATHER_HAIL                 4
-#define ENUM_WEATHER_SUN_PRIMAL           5
-#define ENUM_WEATHER_RAIN_PRIMAL          6
-#define ENUM_WEATHER_STRONG_WINDS         7
-#define ENUM_WEATHER_SNOW                 8
-#define ENUM_WEATHER_FOG                  9
+#define B_WEATHER_NONE          0
+#define B_WEATHER_RAIN_NORMAL   (1 << 0)
+#define B_WEATHER_RAIN_PRIMAL   (1 << 1)
+#define B_WEATHER_RAIN_DOWNPOUR (1 << 2)  // unused
+#define B_WEATHER_RAIN          (B_WEATHER_RAIN_NORMAL | B_WEATHER_RAIN_PRIMAL | B_WEATHER_RAIN_DOWNPOUR)
+#define B_WEATHER_SANDSTORM     (1 << 3)
+#define B_WEATHER_SUN_NORMAL    (1 << 4)
+#define B_WEATHER_SUN_PRIMAL    (1 << 5)
+#define B_WEATHER_SUN           (B_WEATHER_SUN_NORMAL | B_WEATHER_SUN_PRIMAL)
+#define B_WEATHER_HAIL          (1 << 6)
+#define B_WEATHER_SNOW          (1 << 7)
+#define B_WEATHER_FOG           (1 << 8)
+#define B_WEATHER_STRONG_WINDS  (1 << 9)
+
+#define B_WEATHER_ANY           (B_WEATHER_RAIN | B_WEATHER_SANDSTORM | B_WEATHER_SUN | B_WEATHER_HAIL | B_WEATHER_STRONG_WINDS | B_WEATHER_SNOW | B_WEATHER_FOG)
+#define B_WEATHER_PRIMAL_ANY    (B_WEATHER_RAIN_PRIMAL | B_WEATHER_SUN_PRIMAL | B_WEATHER_STRONG_WINDS)
+
+
+enum BattleWeather
```

Comment:
maybe add a comment that the order of these does not have to match the weather flags. Alternatively, could move the enum above the defines, and do something like `#define B_WEATHER_RAIN_NORMAL   (1 << BATTLE_WEATHER_RAIN)`, though this would make the first bit unused

Reply:
We could remove `BATTLE_WEATHER_NONE` since it is unused I believe so we don't loose a bit.

Reply:
Yeah I think that works!

Reply:
Addressed. I've left the comment out since as I've understood it it was either or